client/app/match/match.js

```diff
@@ -33,82 +25,89 @@ angular.module( 'dinnerDaddy.match', ['dinnerDaddy.services'] )
       currRestaurantIndex++;
       $scope.currRestaurant = $scope.restaurants[currRestaurantIndex];
       var currentImageURL = $scope.currRestaurant.image_url;
-      $scope.currRestaurantImageHD = currentImageURL.slice(0,currentImageURL.length-6) + 'l.jpg'; 
-      console.log('current restaurant: ', $scope.currRestaurant);
+      $rootScope.currRestaurantImageHD = currentImageURL.slice(0,currentImageURL.length-6) + 'l.jpg'; 
   };
 
-  $scope.init = function(location) {        //as soon as the view is loaded request the first movie-package here
-    fetchRestaurants($scope.session.sessionLocation);
+  $scope.init = function (location) {
+    // Get restaurants with location defined by user when session was created
+    fetchRestaurants($rootScope.currentSession.sessionLocation);
   };
 
-  $scope.init();
-
-  // Listen for the signal to redirect to a 'match found' page.
-  Socket.on( 'matchRedirect', function( id ) {
-    // id refers to the id of the movie that the group matched on
-    Match.matchRedirect( id );
-  });
+  // Check if there is a currentSession on rootScope, if not create a new session
+  if (!$rootScope.currentSession) {
+    Session.getSession($window.localStorage.getItem('sessionId'))
+    .then(function (session) {
+      $rootScope.currentSession = session;
+      $scope.init();
+    });
+  } else {
+    $scope.init();
+  }
 
   $scope.yes = function() {
-    console.log('clicked on yes');
-    Match.sendVote($scope.session.sessionName, $scope.user.name, currRestaurantIndex, true )
-    // For every 'yes' we want to double check to see if we have a match. If we do,
-    // we want to send a socket event out to inform the server.
+    Match.sendVote($rootScope.currentSession.sessionName, $scope.user.name, currRestaurantIndex, true, $rootScope.currentSession.id)
```

Comment:
Remove unnecessary variables (seems like we're just using some of these for the socket.emit?)